pkgs/development/python-modules/paddleocr/requirements-txt.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/requirements.txt b/requirements.txt
+index 4aabba9b..19e8e637 100644
+--- a/requirements.txt
++++ b/requirements.txt
+@@ -1,14 +1,10 @@
+ shapely
+ scikit-image
+-imgaug
+ pyclipper
+ lmdb
+ tqdm
+ numpy
+-visualdl
+ rapidfuzz
+-opencv-python==4.6.0.66
+-opencv-contrib-python==4.6.0.66
+ cython
+ lxml
+ premailer
+@@ -16,4 +12,4 @@ openpyxl
+ attrdict
+ Polygon3
+ lanms-neo==1.0.2
+-PyMuPDF<1.21.0
+\ No newline at end of file
++PyMuPDF
```

Comment:
The way to remove dependencies in the requirements.txt is usually to use the `pythonRelaxDepsHook` as a `nativeBuildInput`.
If you look for usage in nixpkgs, you can find `pythonRemoveDeps`.

Reply:
I removed the `requirements-txt.patch` and I added the following:

```nix
{
  nativeBuildInputs = [ pythonRelaxDepsHook ];
  pythonRemoveDeps = [
    "imgaug"
    "visualdl"
    "opencv-python"
    "opencv-contrib-python"
  ];
}
```

...but I do now get the error:

```text
> ERROR: Could not find a version that satisfies the requirement opencv-contrib-python==4.6.0.66 (from paddleocr) (from versions: none)
> ERROR: No matching distribution found for opencv-contrib-python==4.6.0.66
```

What am I missing?

Reply:
Looking at the PR, two packages are missing the format argument. For paddleocr, it should be setuptools i think. Does it still do that if you have the format argument ?

Reply:
I believe `format = "setuptools"` is the default, so things don't change from adding that.

Reply:
It looks like the `pythonRelaxDepsHook` only works for wheel packages?

https://github.com/NixOS/nixpkgs/blob/609397e683536d706b6e054567230a386560dea9/pkgs/development/interpreters/python/hooks/python-relax-deps-hook.sh#L86

Reply:
Ah, i knew it wasnt working with pyproject but thought it was a bit more versatile than that. Lets keep it a patch then

Reply:
This is because the version of the fetched source (rev) does not match the version of the derivation.
The version is defined as 2.7.0.0, but the rev points to 2.6.

Reply:
The wheel referred to here is the build output, and it does not matter whether the format is a wheel or not.

Reply:
Oh. Right. My mistake. I had put version 2.7.0.0 in based on what I saw released on pypi, but later found out that the corresponding source code was not n GitHub (yet), so I switched to 2.6 but forgot to change the version.